src/hooks/useForm.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useMemo } from 'react';
+
+// Interfaces START
+
+interface Validator {
+  (value: string): string;
+}
+
+interface FormInput {
+  field: string;
+  value: string;
+  validators: Validator[];
+}
+interface ParameterFunction<T> {
+  (value: FormInput): T;
+}
+
+interface Dictionary<T> {
+  [key: string]: T;
+}
+
+// Interfaces END
+
+// Hook START
+const useForm = (fieldsSettings: FormInput[]) => {
+  const convertArrayToObject = (
+    dataSource: FormInput[],
+    createKey: ParameterFunction<string>,
+    createValue: ParameterFunction<any>
+  ) =>
+    dataSource.reduce((object: Dictionary<any>, item: FormInput) => {
+      object[createKey(item)] = createValue(item);
+      return object;
+    }, {});
+
+  const calculateErrorsForValue = (
+    value: string,
+    validators: Validator[]
+  ): string[] => {
+    let errors: string[] = [];
+    for (let validator of validators) {
+      let errorMessage: string = validator(value);
+      if (errorMessage) {
+        errors.push(errorMessage);
+      }
+    }
+    return errors;
+  };
+
+  const validatorsByKey: Dictionary<[]> = useMemo(() => {
+    return convertArrayToObject(
+      fieldsSettings,
+      (item) => item.field,
+      (item) => item.validators || []
+    );
+  }, []);
```

Comment:
No veo una pregunta similar así que la hago, Por qué no hay dependencias en este useMemo ? dependés de una función (convertArrayToObject) y de fieldsSettings. Sin eso no sé el useMemor para qué serviría

Reply:
I already changed the use of useMemo to the use of useState